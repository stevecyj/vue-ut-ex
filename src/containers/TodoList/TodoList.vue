<script setup>
import { ref } from "vue";

defineProps({});

const count = ref(0);
</script>

<template>
  <h1 class="mmm nnn">TodoList</h1>
</template>

<style scoped></style>
