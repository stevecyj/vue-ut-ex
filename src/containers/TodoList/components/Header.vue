<script setup>
import { ref, defineEmits, computed, nextTick } from "vue";
import { useStore } from "vuex";

const store = useStore();
defineProps({});

// const inputValue = ref("");
const inputValue = computed(() => {
  return store.getters.inputValue;
});

const emit = defineEmits(["add"]);

const changeInputValue = (value) => {
  store.commit("changeInputValue", value);
};

const addTodoItem = () => {
  // emit add event with inputValue
  if (!inputValue || inputValue == "") return;
  emit("add", inputValue);
  changeInputValue("");
};
</script>

<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        type="text"
        data-test="header-input"
        :value="inputValue"
        @input="(e) => changeInputValue(e.target.value)"
        @keyup.enter="addTodoItem"
        placeholder="Please input your todo item !"
      />
    </div>
  </div>
</template>

<style lang="scss" scoped>
.header {
  line-height: 60px;
  background: #333;
  color: #fff;
  font-size: 24px;

  &-content {
    width: 600px;
    margin: 0 auto;
  }
  &-input {
    float: right;
    width: 360px;
    margin-top: 16px;
    line-height: 24px;
    outline: none;
    color: #333;
    text-indent: 10px;
  }
}
</style>
