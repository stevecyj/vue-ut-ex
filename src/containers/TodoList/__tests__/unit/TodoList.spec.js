// import HelloWorld from "./HelloWorld.vue";
import TodoList from "@/containers/TodoList/TodoList.vue";
import Header from "@/containers/TodoList/components/Header.vue";
import { shallowMount } from "@vue/test-utils";

describe("TodoList", () => {
  it("TodoItem 初始化時，undoList 為空 array", () => {
    const wrapper = shallowMount(TodoList);
    const undoList = wrapper.vm.undoList;
    expect(undoList).toEqual([]);
  });

  it("TodoList 監聽到 Header 的 add 事件時，undoList 增加一筆資料", () => {
    const wrapper = shallowMount(TodoList);
    const header = wrapper.findComponent(Header);
    const content = "123";
    header.vm.$emit("add", content);
    const undoList = wrapper.vm.undoList;
    expect(undoList).toEqual([content]);
  });
});
