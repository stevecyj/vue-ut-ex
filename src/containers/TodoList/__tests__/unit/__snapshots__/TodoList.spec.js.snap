// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders properly 1`] = `<h1 class="mmm nnn">TodoList</h1>`;
